autolens/legacy/imaging/model/analysis.py

```diff
@@ -0,0 +1,258 @@
+        """
+        preloads = self.preloads if preload_overwrite is None else preload_overwrite
+
+        return FitImaging(
+            dataset=self.dataset,
+            tracer=tracer,
+            hyper_image_sky=hyper_image_sky,
+            hyper_background_noise=hyper_background_noise,
+            use_hyper_scaling=use_hyper_scaling,
+            settings_pixelization=self.settings_pixelization,
+            settings_inversion=self.settings_inversion,
+            preloads=preloads,
+            profiling_dict=profiling_dict,
+        )
+
+    def visualize(
+        self,
+        paths: af.DirectoryPaths,
+        instance: af.ModelInstance,
+        during_analysis: bool,
+    ):
+        """
+        Output images of the maximum log likelihood model inferred by the model-fit. This function is called throughout
+        the non-linear search at regular intervals, and therefore provides on-the-fly visualization of how well the
+        model-fit is going.
+
+        The visualization performed by this function includes:
+
+        - Images of the best-fit `Tracer`, including the images of each of its galaxies.
+
+        - Images of the best-fit `FitImaging`, including the model-image, residuals and chi-squared of its fit to
+          the imaging data.
+
+        - The hyper-images of the model-fit showing how the galaxies are used to represent different galaxies in
+          the dataset.
+
+        - If hyper features are used to scale the noise or background sky, a `FitImaging` with these features turned
+          off may be output, to indicate how much these features are altering the dataset.
+
+        The images output by this function are customized using the file `config/visualize/plots.ini`.
+
+        Parameters
+        ----------
+        paths
+            The PyAutoFit paths object which manages all paths, e.g. where the non-linear search outputs are stored,
+            visualization, and the pickled objects used by the aggregator output by this function.
+        instance
+            An instance of the model that is being fitted to the data by this analysis (whose parameters have been set
+            via a non-linear search).
+        during_analysis
+            If True the visualization is being performed midway through the non-linear search before it is finished,
+            which may change which images are output.
+        """
+
+        if os.environ.get("PYAUTOFIT_TEST_MODE") == "1":
+            return
+
+        instance = self.instance_with_associated_adapt_images_from(instance=instance)
+
+        fit = self.fit_imaging_via_instance_from(instance=instance)
+
+        if self.positions_likelihood is not None:
+            self.positions_likelihood.output_positions_info(
+                output_path=paths.output_path, tracer=fit.tracer
+            )
+
+        if fit.inversion is not None:
+            try:
+                fit.inversion.reconstruction

```

Comment:
If reconstruction could raise an exception it might be better to make it a method rather than a property; as it stands it looks like this should have no effect